### DISCUSSION TITLE
Dekadal timeframe

### DISCUSSION THREAD

**Discussion by martinedoesgis** [worldbank/GEE_Zonal#4] [Show and tell]
Hello,

I've been asked to produce dekadal rainfall data. It's an odd but apparently common timeframe especially for meteorological analysis, calculated like this: every month has three dekads, such that the first two dekads have 10 days (i.e., 1-10, 11-20), and the third is comprised of the remaining days of the month. See : [http://iridl.ldeo.columbia.edu/maproom/Food_Security/Locusts/Regional/Dekadal_Rainfall/index.html?Set-Language=en](http://iridl.ldeo.columbia.edu/maproom/Food_Security/Locusts/Regional/Dekadal_Rainfall/index.html?Set-Language=en). The main added value is that the same dekad for different years will always be the same days in the calendar (except for end of February), unlike for weeks as week 1 doesn't always start on Jan 1st. 

Coding it directly in Google Earth Engine was quite difficult so I made a small script using Moment.js to produce the list of the dekads for a given timeframe, see here : [https://jsfiddle.net/38bL1q09/3/](https://jsfiddle.net/38bL1q09/3/).  Let me know if/where I can store it on the GitHub for archiving.

It creates in the console a JSON object (that can be converted to CSV and used in Google Earth Engine as a table to loop in) with the following fields:

- **id** : the id of the dekad in the format YYYY-dekadNumber (from 1 to 36)
- **sd** : start date of the dekad in the format YYYY-MM-dd
- **ed** : end date of the dekad in the format YYYY-MM-dd
- **nd** : number of days in the dekad. Always 10 for the two firsts of the month but the third one can be 8,9,10 or 11

Example:
```
    {
        "id": "1981-1",
        "sd": "1981-01-01",
        "ed": "1981-01-10",
        "nd": 10
    },
    {
        "id": "1981-2",
        "sd": "1981-01-11",
        "ed": "1981-01-20",
        "nd": 10
    },
    {
        "id": "1981-3",
        "sd": "1981-01-21",
        "ed": "1981-01-31",
        "nd": 11
    },
    {
        "id": "1981-4",
        "sd": "1981-02-01",
        "ed": "1981-02-10",
        "nd": 10
    }
```

**Comment by andresfchamorro** [comment]
Nice @martinedoesgis , that's definitely useful and good to archive for future reference. I would suggest creating a new folder called /implementations and saving the .js script there.

**Reply by martinedoesgis** [reply]
Ok thanks, I just made a pull request.

**Reply by martinedoesgis** [reply]
Added also my other scripts for archiving, for now in JS but I'll have a look at how to add **dekadal** as a choice for **freq** in **ZonalStats()** in Python.
